lib/promises.helper.js

```diff
@@ -43,13 +43,25 @@ function handlePromise(parameterIndex, original) {
     return fn;
 }
 
-prototype = jasmine.Env.prototype;
+if (Object.keys(jasmine.Env.prototype).length === 0) {
+    version = 2;
+    prototype = jasmine.getEnv();
+} else {
+    version = 1;
+    prototype = jasmine.Env.prototype;
+}
```

Comment:
Would rather use feature detection and not version detection.

```
if (typeof jasmine.getEnv === "function") {
    prototype = jasmine.getEnv();
} else {
    prototype = jasmine.Env.prototype;
}
```

And then later ...

```
// Copy the below code for the afterEach and beforeEach as well to keep them consistent
// but be careful and change the "1" below to the "0" in the handlePromise() call.
[
    "fit",
    "iit",
    "it",
    "xit"
].forEach(function (method) {
    if (typeof prototype[method] === "function") {
        prototype[method] = handlePromise(1, prototype[method]);
    }
});
```